logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -12,7 +12,7 @@ def initialize(logger, output_class, metric, strategy_registry, plugin_args)
     @output_class = output_class
     @metric = metric
     @plugin_args = plugin_args
-    @strategy_registry ||= ::LogStash::OutputDelegatorStrategyRegistry.instance
+    @strategy_registry = strategy_registry || ::LogStash::OutputDelegatorStrategyRegistry.instance
```

Comment:
That is a good call, but I really don't want to have a default value for `plugin_args`. Thinking it through, I'll just remove the `||`. Tests can just be forced to pass a registry.